api/tools/docker-compose.yml

```
@@ -9,17 +9,29 @@ services:
       - 27017:27017
     volumes:
       - db-data:/data
+    healthcheck:
+      test: ["CMD", "mongosh", "--eval", "db.adminCommand('ping')"]
+      interval: 10s
+      timeout: 5s
+      retries: 5
+
   setup:
     image: mongo
     depends_on:
-      - db
+      db:
+        condition: service_healthy
     restart: on-failure
```

```suggestion
    restart: on-failure
    # This will try to initiate the replica set, until it succeeds twice (i.e. until the replica set is already initialized)
```
I think this is weird enough to be worth explaining.